docs/resources/samples/range-basics.md

```diff
@@ -0,0 +1,280 @@
+    if (usedRange) {
+      startCell = usedRange.getLastRow().getCell(0, 0).getOffsetRange(1, 0);
+    } else {
+      startCell = sheet.getRange('A1');
+    }
+    console.log(startCell.getAddress());
+    const targetRange = startCell.getResizedRange(0, data.length - 1);
+    targetRange.setValues([data]);
+    return;
+}
+```
+
+### Color each cell with unique color
+
+```ts
+/**
+ * This sample demonstrates how to iterate over a selected range and set cell property.
+   It colors each cell within the selected range with a random color.
+ */
+function main(workbook: ExcelScript.Workbook) {
+
+    const syncStart = new Date().getTime();
+    // Get selected range
+    const range = workbook.getSelectedRange();
+    const rows = range.getRowCount();
+    const cols = range.getColumnCount();
+    console.log("Start");
+
+    // Color each cell with random color.
+    for (let row = 0; row < rows; row++) {
+        for (let col = 0; col < cols; col++) {
+            range
+                .getCell(row, col)
+                .getFormat()
+                .getFill()
+                .setColor(`#${Math.random().toString(16).substr(-6)}`);
+        }
+    }
+
+    console.log("End");
+    const syncEnd = new Date().getTime();
+    console.log("Completed, took: " + (syncEnd - syncStart) / 1000 + " Sec");
+}
+```
+
+### Update range with values using 2D array
+
+Dynamically calculates the range dimension to update based on 2D array values.
+
+```ts
+function main(workbook: ExcelScript.Workbook) {
+  const currentCell = workbook.getActiveCell();
+  let inputRange = computeTargetRange(currentCell, DATA);
+  // Set range values.
+  console.log(inputRange.getAddress());
+  inputRange.setValues(DATA);
+  // Call a helper function to place border around the range.
+  borderAround(inputRange);
+}
+
+/**
+ * A Helper function that computes the target range given the target range's starting cell and selected range. 
+ */
+function computeTargetRange(targetCell: ExcelScript.Range, data: string[][]): ExcelScript.Range {
+  const targetRange = targetCell.getResizedRange(data.length - 1, data[0].length - 1);
+  return targetRange;
+}
+
+/**
+ * A Helper function that places a border around the range.
+ */
+function borderAround(range: ExcelScript.Range): void {
+  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeLeft).setStyle(ExcelScript.BorderLineStyle.dash);
+  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeRight).setStyle(ExcelScript.BorderLineStyle.dash);
+  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeTop).setStyle(ExcelScript.BorderLineStyle.dash);
+  range.getFormat().getRangeBorder(ExcelScript.BorderIndex.edgeBottom).setStyle(ExcelScript.BorderLineStyle.dash);
+  return;
+}
+
+// Values used for range setup.
+const DATA = [
+  ['Item', 'Bread', 'Donuts', 'Cookies', 'Cakes', 'Pies'],
+  ['Amount', '2', '1.5', '4', '12', '26']
+]
+```
+
+## Methods that return some range metadata
```

Comment:
We should consider pulling this list into the main Object Model article.

Reply:
Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)

Reply:
Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)

Reply:
I feel the "Methods that ..." section would be hard to maintain as we add more methods. Unless you see a lot of value in it, I'd remove it.